lib/components/SelectionControl/radio/radio-button/index.tsx

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import clsx from "clsx";
+import {
+    LabelPairedCircleInfoMdRegularIcon,
+    LabelPairedCircleInfoSmRegularIcon,
+    StandaloneCircleDotBoldIcon,
+    StandaloneCircleRegularIcon,
+} from "@deriv/quill-icons";
+import { Text } from "../../../Typography";
+import "./radio-button.scss";
+import { TMediumSizes } from "../../../../types";
+
+interface IRadio {
+    className?: string;
+    classNameInfo?: string;
+    classNameLabel?: string;
+    defaultChecked?: boolean;
+    disabled?: boolean;
+    hasInfo?: boolean;
+    id?: string;
+    name?: string;
+    onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+    required?: boolean;
+    value?: string;
+    size?: TMediumSizes;
+}
+
+const RadioButton = ({
+    children,
+    className,
+    classNameInfo,
+    classNameLabel,
+    defaultChecked,
+    disabled = false,
+    hasInfo,
+    id,
+    onChange,
+    size = "md",
+    value,
+    ...otherProps
+}: React.PropsWithChildren<IRadio>) => {
+    const [checked, setChecked] = React.useState(defaultChecked);
+
+    React.useEffect(() => {
+        setChecked(defaultChecked);
+    }, [defaultChecked]);
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setChecked(e.target.checked);
+        onChange?.(e);
+    };
+
+    const handleIconClick = (
+        e:
+            | React.MouseEvent<HTMLSpanElement>
+            | React.KeyboardEvent<HTMLSpanElement>,
+    ) => {
+        if (!disabled) {
+            const synthesizedEvent = new Event("change");
+            const input = e.currentTarget.previousSibling as HTMLInputElement;
+            input.dispatchEvent(synthesizedEvent);
+        }
+    };
+
+    const getIcon = () => {
+        if (checked) {
+            return (
+                <StandaloneCircleDotBoldIcon
+                    data-testid={`dt_checked_icon_${value}_${disabled}`}
+                    fill={disabled ? "#b8b8b8" : "#000000"}
+                    iconSize={size}
```

Comment:
Why this icon doesn't have the `tabIndex` and the other one has?

Reply:
I want to avoid selecting the selected option. Since there's no point to click the option which is already selected, I only added tabIndex to the icon which is used for non-selected options.